src/main/java/com/aws/greengrass/mqttbridge/clients/MQTTClient.java

```diff
@@ -81,37 +81,38 @@ public void deliveryComplete(IMqttDeliveryToken token) {
     };
 
     /**
-     * Ctr for MQTTClient.
+     * Construct an MQTTClient.
      *
-     * @param topics             topics passed in by Nucleus
+     * @param brokerUri          broker uri
+     * @param clientId           client id
      * @param mqttClientKeyStore KeyStore for MQTT Client
      * @param executorService    Executor service
      * @throws MQTTClientException if unable to create client for the mqtt broker
      */
-    @Inject
-    public MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore,
+    public MQTTClient(URI brokerUri, String clientId, MQTTClientKeyStore mqttClientKeyStore,
                       ExecutorService executorService) throws MQTTClientException {
-        this(topics, mqttClientKeyStore, executorService, null);
+        this(brokerUri, clientId, mqttClientKeyStore, executorService, null);
         try {
             this.mqttClientInternal = new MqttClient(brokerUri.toString(), clientId, dataStore);
         } catch (MqttException e) {
             throw new MQTTClientException("Unable to create an MQTT client", e);
         }
     }
 
-    protected MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore,
-                         ExecutorService executorService, IMqttClient mqttClient) throws MQTTClientException {
-        try {
-            this.brokerUri = BridgeConfig.getBrokerUri(topics);
-        } catch (URISyntaxException e) {
-            throw new MQTTClientException("Invalid brokerUri configuration", e);
-        }
+    protected MQTTClient(URI brokerUri, String clientId, MQTTClientKeyStore mqttClientKeyStore,
+                         ExecutorService executorService, IMqttClient mqttClient) {
+        Objects.requireNonNull(brokerUri);
+        Objects.requireNonNull(clientId);
+        this.brokerUri = brokerUri;
+        this.clientId = clientId;
         this.mqttClientInternal = mqttClient;
         this.dataStore = new MemoryPersistence();
-        this.clientId = BridgeConfig.getClientId(topics);
-        this.mqttClientKeyStore = mqttClientKeyStore;
-        this.mqttClientKeyStore.listenToUpdates(updateListener);
         this.executorService = executorService;
+        this.mqttClientKeyStore = mqttClientKeyStore;
+        this.ssl = "ssl".equalsIgnoreCase(brokerUri.getScheme());
```

Comment:
MQTTBridge: `this.ssl = brokerUri.getScheme().equalsIgnoreCase("ssl");`
MQTTClient: `this.ssl = "ssl".equalsIgnoreCase(brokerUri.getScheme());`

Either is fine but not both. Creates inconsistency.

Reply:
yep, thanks for the catch!

Reply:
`this.ssl = "ssl".equalsIgnoreCase(brokerUri.getScheme());`
JFYI, the above pattern is preferred since it avoids a potential null exception.